nucliadb/nucliadb/ingest/orm/processor/__init__.py

```diff
@@ -317,18 +320,21 @@ async def _mark_resource_error(
                 f"Skip when resource does not even have basic metadata: {resource}"
             )
             return
-        txn = None
+        prev_txn = resource.txn
         try:
             async with self.driver.transaction() as txn:
+                resource.txn = txn
                 resource.basic.metadata.status = PBMetadata.Status.ERROR
-                await set_basic(txn, resource.kb.kbid, resource.uuid, resource.basic)
+                await resource.set_basic(resource.basic)
                 await txn.commit()
 
             await shard.add_resource(
```

Comment:
This in combination with the `resource.clean()` from above was the one creating the empty broker message